javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +519,371 @@ object Character {
+    val TRANSPORT_AND_MAP_SYMBOLS = addUnicodeBlock("TRANSPORT_AND_MAP_SYMBOLS", "Transport and Map Symbols", 0x1F680, 0x1F6FF)
+    val ALCHEMICAL_SYMBOLS = addUnicodeBlock("ALCHEMICAL_SYMBOLS", "Alchemical Symbols", 0x1F700, 0x1F77F)
+    val CJK_UNIFIED_IDEOGRAPHS_EXTENSION_B = addUnicodeBlock("CJK_UNIFIED_IDEOGRAPHS_EXTENSION_B", "CJK Unified Ideographs Extension B", 0x20000, 0x2A6DF)
+    val CJK_UNIFIED_IDEOGRAPHS_EXTENSION_C = addUnicodeBlock("CJK_UNIFIED_IDEOGRAPHS_EXTENSION_C", "CJK Unified Ideographs Extension C", 0x2A700, 0x2B73F)
+    val CJK_UNIFIED_IDEOGRAPHS_EXTENSION_D = addUnicodeBlock("CJK_UNIFIED_IDEOGRAPHS_EXTENSION_D", "CJK Unified Ideographs Extension D", 0x2B740, 0x2B81F)
+    val CJK_COMPATIBILITY_IDEOGRAPHS_SUPPLEMENT = addUnicodeBlock("CJK_COMPATIBILITY_IDEOGRAPHS_SUPPLEMENT", "CJK Compatibility Ideographs Supplement", 0x2F800, 0x2FA1F)
+    val TAGS = addUnicodeBlock("TAGS", "Tags", 0xE0000, 0xE007F)
+    val VARIATION_SELECTORS_SUPPLEMENT = addUnicodeBlock("VARIATION_SELECTORS_SUPPLEMENT", "Variation Selectors Supplement", 0xE0100, 0xE01EF)
+    val SUPPLEMENTARY_PRIVATE_USE_AREA_A = addUnicodeBlock("SUPPLEMENTARY_PRIVATE_USE_AREA_A", "Supplementary Private Use Area-A", 0xF0000, 0xFFFFF)
+    val SUPPLEMENTARY_PRIVATE_USE_AREA_B = addUnicodeBlock("SUPPLEMENTARY_PRIVATE_USE_AREA_B", "Supplementary Private Use Area-B", 0x100000, 0x10FFFF)
+
+    ////////////////
+    // End Generated
+    ////////////////
+    // scalastyle:on line.size.limit
+
+    // Additional JVM Aliases
+    blocksByNormalizedName.put("Greek and Coptic".toLowerCase, GREEK)
+    blocksByNormalizedName.put("GreekandCoptic".toLowerCase, GREEK)
+    blocksByNormalizedName.put("GREEK_AND_COPTIC".toLowerCase, GREEK)
+    blocksByNormalizedName.put("TaiLe".toLowerCase, TAI_LE)
+    blocksByNormalizedName.put("Combining Diacritical Marks for Symbols".toLowerCase, COMBINING_MARKS_FOR_SYMBOLS)
+    blocksByNormalizedName.put("CombiningDiacriticalMarksforSymbols".toLowerCase, COMBINING_MARKS_FOR_SYMBOLS)
+
+    def forName(blockName: String): UnicodeBlock = {
+      val key: String = blockName.toLowerCase
+      val block = blocksByNormalizedName.get(key)
+      if (block == null) throw new IllegalArgumentException()
+      block
+    }
+
+    def of(c: scala.Char): UnicodeBlock = of(c.toInt)
+
+    def of(codePoint: scala.Int): UnicodeBlock = {
+      // scalastyle:off return
+      if (!Character.isValidCodePoint(codePoint)) {
+        throw new IllegalArgumentException()
+      }
+
+      var low: Int = 0
+      var mid: Int = -1
+      var high: Int = allBlocks.size() - 1
+
+      while (low <= high) {
+        mid = (low + high) >>> 1
+        val block = allBlocks.get(mid)
+        if (codePoint > block.end) {
+          low = mid + 1
+        } else if (codePoint >= block.start) {
+          return block
+        } else {
+          high = mid - 1
+        }
+      }
+
+      null
```

Comment:
Use `Collections.binarySearch`?

Reply:
I thought about this, but didn't explore it.  I don't think the signatures match up, since , with the current collection I'm using, but perhaps can instead leverage something like `charTypeIndices`, and have the ending codepoint in the array doing the searching for the index in `allBlocks` is that generally what you mean/are thinking about?

Reply:
Nah, I just saw a binary search and though "that must be implemented somewhere else" and it is... but because it uses `Comparator` it requires symmetric comparison :-/ So no, you can't really apply it directly.